src/main/java/com/cgdpd/library/entity/TrackingStatus.java

```diff
@@ -0,0 +1,62 @@
+package com.cgdpd.library.entity;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public enum TrackingStatus {
+    BEING_PROCESSED("Being Processed"),
+    AVAILABLE("Available"),
+    ON_HOLD("On Hold"),
+    CHECKED_OUT("Checked Out"),
+    LOST("Lost"),
+    RETIRED("Retired"),
+    REFERENCE("Reference");
+
+    private final String status;
+    private final Map<TrackingStatus, Boolean> nextStatuses;
```

Comment:
Let's make this property static where key is tracking status and value is a Set of the possible status transitions